File: backend/Services/Parsers/FramedParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class FramedParser : BasicScoreResultParser
    {
        private ILogger<FramedParser> _logger;

        public FramedParser(ILogger<FramedParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override DateTime ActiveAfter => DateTime.MinValue;
        public override string Category => "Media";
        public override string GameName => "Framed";
        public override string? HelpText => null;
        // \uD83C\uDFA5 = 🎥
        // \uD83D\uDFE9 = 🟩
        // \uD83D\uDFE8 = 🟨
        // \uD83D\uDFE5 = 🟥
        // \u2B1C\uFE0F = ⬜️
        // \u2B1B\uFE0F = ⬛️
        protected override Regex Parser => new Regex($"{GameName} #\\d+\n\uD83C\uDFA5(?<{ScoreGroup}>[^\n]*)", RegexOptions.Singleline);
        protected override string? ExtraContent => Url;
        public override string Url => "https://framed.wtf";

        protected override DailyResult SetScore(DailyResult dailyResult, Match parserResults)
        {
            if (!parserResults.Groups.ContainsKey(ScoreGroup))
            {
                return dailyResult;
            }

            var score = parserResults.Groups[ScoreGroup].Value.Trim().Replace(" ", string.Empty);

            var successIndex = score.IndexOf("🟩");
            if (successIndex > -1)
            {
                dailyResult.Score = (successIndex + "🟩".Length) / 2;
            }

            return dailyResult;
        }
    }
}

File: lambdas/src/parsers/framed-parser.ts
import { DailyResult } from '../models/daily-result';
import { BasicScoreResultParser } from './base/basic-score-parser';
import { getGroup, hasGroup } from './base/result-parser';

export class FramedParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Framed';
  readonly helpText = null;
  readonly url = 'https://framed.wtf';
  protected readonly extraContent = this.url;
  
  // Captures emoji line after 🎥: "Framed #1064\n🎥 🟥 🟥 🟥 🟥 🟥 🟩"
  protected readonly parser = new RegExp(
    `${this.gameName} #\\d+\\n🎥(?<${BasicScoreResultParser.SCORE_GROUP}>[^\\n]*)`,
    's'
  );
  
  protected setScore(dailyResult: DailyResult, match: RegExpMatchArray): DailyResult {
    if (!hasGroup(match, BasicScoreResultParser.SCORE_GROUP)) {
      return dailyResult;
    }
    
    const scoreStr = getGroup(match, BasicScoreResultParser.SCORE_GROUP);
    if (!scoreStr) return dailyResult;
    
    // Remove spaces from the emoji line
    const score = scoreStr.trim().replace(/ /g, '');
    
    // Find position of 🟩 (success)
    const successIndex = score.indexOf('🟩');
    if (successIndex > -1) {
      // Each emoji is 2 chars, so divide by 2 to get position (1-6)
      dailyResult.score = (successIndex + '🟩'.length) / 2;
    }
    
    return dailyResult;
  }
}
